lib/pbio/sys/command.c

```diff
@@ -48,6 +60,15 @@ pbio_pybricks_error_t pbsys_command(const uint8_t *data, uint32_t size) {
             #endif
             // If no consumers are configured, goes to "/dev/null" without error
             return PBIO_PYBRICKS_ERROR_OK;
+        case PBIO_PYBRICKS_COMMAND_WRITE_PROGRAM_DATA_BUFFER:
+            if (write_program_data_buffer_callback && size > 3) {
+                uint16_t offset = pbio_get_uint16_le(&data[1]);
+                uint16_t data_size = size - 3;
+                const uint8_t *data_write = &data[3];
+                write_program_data_buffer_callback(offset, data_size, data_write);
```

Comment:
Why not return an error code here so we can notify the sender if they sent invalid data (e.g. too big offset)?

Reply:
That could work, although I think success would be OK if the program isn't running, so we don't throw errors if a simple web app keeps sending all the time.